### ISSUE TITLE
:arg2-extend + :list/respect-nl? exception

### ISSUE THREAD

**Issue by atdixon** [kkinnear/zprint#200]
Reproduced with `1.1.1` and `1.1.2`:

```
;; repro.clj
(ns repro)

(deffoo Xyz []
  (go []))
```

`./zprintm-1.1.2 '{:list {:respect-nl? true} :fn-map {"deffoo" :arg2-extend}}' < src/repro_1.clj`

> Failed to zprint: java.lang.Exception: count-newline-types: more than one type or wrong type! count:6 style-vec:[["\n  " :none :newline 2] ["\n    " :none :indent 1] ["(" :none :left] ["go" :none :element] [" " :none :whitespace 6] ["[" :none :left] ["]" :none :right 8] [")" :none :right 4]]

**Comment by atdixon** [comment]
Realizing now that this `deffoo` isn't in the spirit of "extend", where a protocol is typically expected before the function impl.

zprint does succeed with `IFace` in the form.

```
(ns repro)

(deffoo Xyz []
  IFace
  (go []))
```

So feel free to close this as unsupported. Or perhaps zprint could be more robust here instead of throwing an exception ...?

**Comment by kkinnear** [comment]
Thanks for noticing both the problem and that the failing form isn't quite in the spirit of "extend"!  I definitely need to figure out what I was thinking with this code.

Right before it throws this failure, I have:

`; TODO: Take this out if we don't get any exceptions while testing.`

I totally agree that it makes sense to not throw an exception here.  I need to figure out why I ever thought that it might matter what kind of newlines I had and whether they needed to come from the same place.  But I will certainly fix this to be more robust!

Thanks again for noticing this!

**Comment by kkinnear** [comment]
This is fixed in `1.2.0`.
